### ISSUE TITLE
How to use cri-o as runtime on k3s 1.20+ version

### ISSUE THREAD

**Issue by AbelTaylor** [k3s-io/k3s#2741]
as the link https://rancher.com/docs/k3s/latest/en/installation/install-options/server-config/#agent-runtime

According to this introduction, it seems that the only runtimes we can choose from are docker and containerd, but 1.20 is no longer recommended to continue to use docker, and there are some inappropriate use of containerd in our environment. Therefore, we want to use cri-o or podman as Runtime is used for testing, how do I configure it?

**Comment by brandond** [comment]
Containerd and docker are the only options at the moment. Note that while Podman shares on-disk datastores with CRI-O it is not itself a CRI.

Can you explain what you mean by "inappropriate use of containerd"?

**Comment by erikwilson** [comment]
I feel like at some point when playing around with SELinux that I was able to use CRI-O by setting the `--container-runtime-endpoint` flag to an existing CRI-O instance.

**Comment by myzhuh** [comment]
hi,brandond
> Containerd and docker are the only options at the moment. Note that while Podman shares on-disk datastores with CRI-O it is not itself a CRI.
> 
> Can you explain what you mean by "inappropriate use of containerd"?

Containerd's support for private registry does not seem to be very good, and the speed of pulling images on dockerhub is very slow. We tried to configure according to the official recommendations of k3s, but the effect has not been very good.

**Comment by brandond** [comment]
Many of the earlier issues with private registry have been resolved, and its is now pretty easy to configure via the K3s registries.yaml.

With regards to image pull speed, I don't think a different container runtime should make much of a difference. One of the things that we do for containerd that might make a difference is that we set `--serialize-image-pulls=false` which may have a side-effect of causing bandwidth and disk IO spikes when pulling multiple images at the same time.
